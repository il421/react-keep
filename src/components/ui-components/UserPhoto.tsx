import React from "react";

interface UserPhotoProps
  extends Pick<HTMLImageElement, "src" | "width" | "height"> {
  className?: string;
}

export class UserPhoto extends React.PureComponent<UserPhotoProps> {
  render() {
    const { className, ...rest } = this.props;
    return (
      <div className={className}>
        <img {...rest} alt="User photo" />
      </div>
    );
  }
}
