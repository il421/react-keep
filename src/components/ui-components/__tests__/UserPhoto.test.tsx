import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { UserPhoto, UserPhotoProps } from "../UserPhoto";

let props: UserPhotoProps, wrapper: ShallowWrapper;

beforeEach(() => {
  props = {
    src: "src",
    height: 100,
    width: 100
  };
});

test("should render UserPhoto", () => {
  wrapper = shallow(<UserPhoto {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
