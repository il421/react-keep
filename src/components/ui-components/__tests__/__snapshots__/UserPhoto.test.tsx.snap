// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserPhoto 1`] = `
<div>
  <img
    alt="User photo"
    height={100}
    src="src"
    width={100}
  />
</div>
`;
